### PR TITLE
Add option to make segments visible in inactive windows

### LINKED ISSUE #587: Option to make segments visible in inactive windows
First of all, thank you all for your work on `doom-modeline`. 
Much appreciated, `doom-modeline` is great.

**Is your feature request related to a problem? Please describe.**

Some segments are displayed only in active windows. Personally, I tested and am interested in `mu4e` segment but looking at the code, this should be the case for `gnus`, `irc`, `battery`, etc as well. I believe this behavior applies to all segments that start with  `(when (and (doom-modeline--active)`.

I believe this works perfectly for many users, and I can imagine wanting the same behavior on a small laptop screen or when using Emacs inside of a small window. But it IMHO falls short when you have a larger screen estate. My standard situation is two fullscreen Emacs frames split into ~8 windows. I don't want to clutter my active window modeline with all the not-immediately-needed information such as the number of unread emails, the number of unread RSS articles, IRC buffers, etc when there is a lot of other windows visible.

**Describe the solution you'd like**

I would like to display my `mu4e` segment in my `mu4e-headers` window, `irc-buffers` segment in my `irc-channel-mode` windows, and so on. This can easily be done via `doom-modeline-mode-alist` but the segments are displayed only for active windows. I want to display them for inactive windows as well.

**Describe alternatives you've considered**

I considered overriding `(doom-modeline--active)` to always return `t` and go from there. But that's just too hacky and probably will break many things, and I would like to believe that I am not the only one wanting this feature.

**Additional context**

I am imagining the following implementation but feel free to completely ignore it if you prefer something different.

We could add `doom-modeline-always-visible` variable that would be user-defined. In my case I would do e.g. `(setq doom-modeline-always-visible '(mu4e irc-buffers))`. And we would change all the segments to check `(or (doom-modeline--active) (member 'mu4e doom-modeline-always-visible))` instead of current `(doom-modeline--active)`.

What do you think?

### PR DESCRIPTION
**Pull request by FrostyX**
Fix #587

### CODE REVIEW

**Comment by FrostyX** [comment]
I didn't update all segments and their occurrences of `(doom-modeline--active)`, only those that also check `(not doom-modeline--limited-width-p)`. I was thinking that I could move the `(not doom-modeline--limited-width-p)` into `doom-modeline--segment-visible` function as well.

What do you think?

File: doom-modeline-core.el [hunk 1]
```diff
@@ -539,6 +539,12 @@ If nil, display only if the mode line is active."
   :type 'boolean
   :group 'doom-modeline)
 
+(defcustom doom-modeline-always-visible-segments nil
+  "A list of segments that should be visible even in
+inactive windows."
+  :type '(repeat string)
```
Comment (seagle0128) [inline doom-modeline-core.el:545]: Why is `string` type used here? May `symbol` be convient?
Anyway, what ever type, it's better provide an example in docstring and/or README.

**Review by seagle0128 — CHANGES_REQUESTED** [review]

**Comment by seagle0128** [comment]
> I didn't update all segments and their occurrences of `(doom-modeline--active)`, only those that also check `(not doom-modeline--limited-width-p)`. I was thinking that I could move the `(not doom-modeline--limited-width-p)` into `doom-modeline--segment-visible` function as well.
> 
> What do you think?

Yeah, it seems `doom-modeline--limited-width-p` is tightly coupled, and could be moved into the check function.
And I noticed some are missing, e.g. `buffer-position` segment. You also need to check `active` (equals to `(doom-modeline--active)`)  😄

File: doom-modeline-core.el (hunk 1 above)
Comment (FrostyX) [reply doom-modeline-core.el:545]: You are correct, thank you

**Comment by FrostyX** [comment]
> Yeah, it seems doom-modeline--limited-width-p is tightly coupled, and could be moved into the check function.

Okay :-)

> And I noticed some changes are missing, e.g. buffer-position segment. 

I left them intentionally because they don't check for `(not doom-modeline--limited-width-p)`. Therefore, by moving the check into the `doom-modeline--segment-visible` we would start checking it now. Do we want to do that?

> You also need to check active (equals to (doom-modeline--active)) 

Can you elaborate, please?
As I understand it, both `doom-modeline--active` and `doom-modeline--segment-visible` return a boolean and I swapped one for another, so I don't see any need for `equals`. But I might be missing something.

**Comment by seagle0128** [comment]
Please see the snippet below.

```elisp
(and active
     (not doom-modeline--limited-width-p)
```
should equals to `doom-modeline--segment-visible`, right?

```elisp
     (cond ((and active
                 (bound-and-true-p nyan-mode)
                 (not doom-modeline--limited-width-p)
                 (>= (window-width) nyan-minimum-window-width))
            (concat
             doom-modeline-wspc
             (propertize (nyan-create) 'mouse-face mouse-face)))
           ((and active
                 (bound-and-true-p poke-line-mode)
                 (not doom-modeline--limited-width-p)
                 (>= (window-width) poke-line-minimum-window-width))
            (concat
             doom-modeline-wspc
             (propertize (poke-line-create) 'mouse-face mouse-face)))
           ((and active
                 (bound-and-true-p mlscroll-mode)
                 (not doom-modeline--limited-width-p)
                 (>= (window-width) mlscroll-minimum-current-width))
            (concat
             doom-modeline-wspc
             (let ((mlscroll-right-align nil))
               (format-mode-line (mlscroll-mode-line)))))
           ((and active
                 (bound-and-true-p sml-modeline-mode)
                 (not doom-modeline--limited-width-p)
                 (>= (window-width) sml-modeline-len))
            (concat
             doom-modeline-wspc
             (propertize (sml-modeline-create) 'mouse-face mouse-face)))
           (t ""))
```

**Comment by FrostyX** [comment]
Oh, I think I understand now. Updated.
I also addressed the points from earlier comments.

PTAL

**Comment by seagle0128** [comment]
Thank you!

**Pull request merged**